viewing_party/party.py

```
@@ -1,23 +1,168 @@
+
+
+TITLE_KEY = "title"
+GENRE_KEY = "genre"
+RATING_KEY = "rating"
+
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {
+        TITLE_KEY: title,
+        GENRE_KEY: genre,
+        RATING_KEY: rating}
+
+    if not new_movie[TITLE_KEY]:
+        return None
+    if not new_movie[GENRE_KEY]:
+        return None
+    if not new_movie[RATING_KEY]:
+        return None
+    return new_movie
+
+
+def add_to_watched(user_data, movie):
+    user_data = {"watched": [movie]}
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data = {"watchlist": [movie]}
+    return user_data
+
+
+def watch_movie(user_data, title):
+    if user_data["watchlist"]:
+        for movie in user_data['watchlist']:
+            if movie['title'] == title:
+                user_data["watched"].append(movie)
+                user_data['watchlist'].remove(movie)
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
 
-# -----------------------------------------
-# ------------- WAVE 3 --------------------
-# -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum = 0
+    ratings_list = []
+    if user_data['watched']:
+        for movie in user_data["watched"]:
+            ratings_list.append(movie['rating'])
+        for movie in ratings_list:
+            sum += movie
+        average = sum/len(ratings_list)
+        return average
+    return 0.0
 
-        
-# -----------------------------------------
-# ------------- WAVE 4 --------------------
-# -----------------------------------------
 
-# -----------------------------------------
-# ------------- WAVE 5 --------------------
-# -----------------------------------------
+def get_most_watched_genre(user_data):
+    genre_list = []
+    genre_tally_dict = {}
+    most_watched_str = None
+    most_watched_int = 0
+    if user_data:

```

As above, I'd leave off some of these safety checks. We expect a valid `user_data` structure to be passed in, so it can be helpful for the code to crash rather than silently continuing on.